### DISCUSSION TITLE
1.1.2

### DISCUSSION THREAD

**Discussion by shuding** [vercel/swr#1743] [New]
### Use the Latest Fetcher Function

SWR will now use the latest fetcher function passed to the hook, when sending the request. Previously it uses the initial snapshotted fetcher.

### Avoid Unnecessary Auto Revalidations

When refocusing on the window with state changes (like clicking a button that changes the SWR key immediately), SWR now avoids revalidations if they're not necessary. Details can be found in #1720.

### New Types for `useSWRInfinite`

Two types for `useSWRInfinite` are added: `SWRInfinteHook` and `SWRInfinteKeyLoader`. You can use them to type the hook and the `getKey` function.

### New `populateCache` option for `mutate`

A new option to control if the mutation data should be written to the cache. You can find the details in #1729.

## What's Changed
* Make polyfill for `requestAnimationFrame` more robust by @thomaspaulmann in https://github.com/vercel/swr/pull/1707
* type(infinite): export SWRInfinteHook and SWRInfiniteKeyLoader from infinite, rename InfiniteFetcher to SWRInfiniteFetcher by @houkanshan in https://github.com/vercel/swr/pull/1714
* feat: Ensure auto revalidations are executed after state updates by @shuding in https://github.com/vercel/swr/pull/1720
* Add moduleNameMapper to jest by @shuding in https://github.com/vercel/swr/pull/1724
* Improve test by @promer94 in https://github.com/vercel/swr/pull/1725
* feat: Use the latest reference of fetcher by @shuding in https://github.com/vercel/swr/pull/1727
* fix: Re-export InfiniteFetcher by @shuding in https://github.com/vercel/swr/pull/1728
* feat: Add `populateCache` option to `mutate` by @shuding in https://github.com/vercel/swr/pull/1729
* Fix local state sharing example deps by @huozhi in https://github.com/vercel/swr/pull/1737
* Merge states for concurrent requests by @shuding in https://github.com/vercel/swr/pull/1741
* Skip error retrying when document is not active and improve tests by @shuding in https://github.com/vercel/swr/pull/1742

## New Contributors
* @thomaspaulmann made their first contribution in https://github.com/vercel/swr/pull/1707
* @houkanshan made their first contribution in https://github.com/vercel/swr/pull/1714

**Full Changelog**: https://github.com/vercel/swr/compare/1.1.1...1.1.2

<hr /><em>This discussion was created from the release <a href='https://github.com/vercel/swr/releases/tag/1.1.2'>1.1.2</a>.</em>

**Comment by joelmoss** [comment]
> Use the Latest Fetcher Function
> SWR will now use the latest fetcher function passed to the hook, when sending the request. Previously it uses the initial snapshotted fetcher.

Just an FYI for the future, but this should have been marked as a breaking change. It broke my app, as I was relying on variables from outside the fetcher function.

Fortunately it turned out to be an easy fix.

**Reply by shuding** [reply]
Can you share a bit more about why it breaks your code instead of fixes it? Why don’t you want the fetcher to use the latest value?

**Comment by joelmoss** [comment]
It was simply because of how I was using variables within the fetcher function. After changing it to not use variables from outside, all is fine.  So the fix was simple, and actually better. But your change broke backwards compat due to the latest fetcher being used instead of the initial.
